File: Trabalhos-Práticos/TP1/Questão 16 - Palindromo Recursivo - Java/PalindromoRec.java
import java.util.Scanner;


public class PalindromoRec {
	public static boolean ehPalindromo(String str,int i) {
		boolean palindromo = true;
		int tamanho = str.length() ;

		if(i>tamanho/2) {
			palindromo = true;
		}
		else {
			if(str.charAt(i) != str.charAt(tamanho - 1 - i)) {
				palindromo = false;
			}else {
				palindromo = ehPalindromo(str, i+1);
			}
		}

		return palindromo;

		
	}
	
	public static boolean ehFim(String string) {
		return (string.length() == 3 && string.charAt(0) == 'F' && string.charAt(1) == 'I' && string.charAt(2) == 'M');
	}

	public static void main(String[] args) {
		Scanner scanner = new Scanner(System.in);
		String entrada = scanner.nextLine();

		while(!ehFim(entrada)) {
			if(ehPalindromo(entrada, 0)) {
				System.out.println("SIM");
			} else {
				System.out.println("NAO");
			}

			entrada = scanner.nextLine();
		}

		scanner.close();
	}
}

	

			
		
			

File: Trabalhos-Práticos/TP1/Questão 17 - Palindromo Reccursivo - C/PalindromoRec.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdbool.h>

#include <locale.h>

#define TAM 100
#define TAM_STR "99"

bool ehPalindromo(char string[], int i) {
	bool palindromo = true;
	int tamanho = strlen(string);

	if(i > tamanho/2) {
		palindromo = true;
	}else {
		if(string[i] != string[tamanho - 1-  i]) {
			palindromo = false;
		} else {
			palindromo = ehPalindromo(string, i + 1);
		}
	}
	return palindromo;
}

bool ehFim(char str[]) {
	bool ehFim = false;
	if(strlen(str) == 3) {
		if(str[0] == 'F' && str[1] == 'I' && str[2] == 'M'){
			ehFim = true;
		}
	}
	return ehFim;
}

int main() {
	setlocale(LC_ALL, "pt_BR.UTF-8");

	char entrada[TAM];
	

	fgets(entrada,sizeof(entrada), stdin);
	entrada[strcspn(entrada, "\n")] = '\0';

	while(!ehFim(entrada)) {
		if(ehPalindromo(entrada,0)) {
			printf("SIM\n");
		} else {
			printf("NAO\n");
		}

		fgets(entrada,sizeof(entrada), stdin);
		entrada[strcspn(entrada, "\n")] = '\0';
	
	}

	return 0;

}



